src/geom/LinearFunctor.cc

```diff
@@ -0,0 +1,50 @@
+// -*- lsst-c++ -*-
+
+/* 
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "boost/make_shared.hpp"
+#include "lsst/afw/geom/Functor.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+LinearFunctor::LinearFunctor(double slope, double intercept) 
+   : Functor("LinearFunctor"), _slope(slope), _intercept(intercept) {
+}
+
+PTR(Functor) LinearFunctor::clone() const {
+   return boost::make_shared<LinearFunctor>(_slope, _intercept);
+}
+
+double LinearFunctor::operator()(double x) const {
+   return _slope*x + _intercept;
+}
+
+double LinearFunctor::derivative(double x) const {
```

Comment:
Sorry I'm kind of late to the party here, but I thought that LinearFunctor is supposed to set a good example, so I would have thought it should have an `inverse()` method since it's definitely analytically invertible.